### PR TITLE
Raise ValueError if TFR freqs <= 0

### PR DESCRIPTION
**Pull request by DiGyt**
Raise a ValueError for invalid frequencies passed into ```tfr_morlet``` and ```tfr_multitaper```, according to #5708 

#### Additional information
Raise a ValueError ```"all computed frequencies in freqs must be greater than 0."``` if the freqs passed to ```mne/time_frequency/tfr.py::morlet``` and ```mne/time_frequency/tfr.py::_make_dpss``` are <= 0.

Note:
Used ```np.any([f <= 0 for f in freqs])``` instead of ```np.any(freqs <= 0)``` to also handle simple lists.

### CODE REVIEW

File: mne/time_frequency/tfr.py [hunk 1]
```diff
@@ -66,6 +66,10 @@ def morlet(sfreq, freqs, n_cycles=7.0, sigma=None, zero_mean=False):
     Ws = list()
     n_cycles = np.atleast_1d(n_cycles)
 
+    if np.any([f <= 0 for f in freqs]):
```
Comment (agramfort) [inline mne/time_frequency/tfr.py]: I would mkae sure freqs is an array and do:

if np.any(freqs <= 0)

list comprehensions are slow and usually avoided in numericla code.

Comment (DiGyt) [reply mne/time_frequency/tfr.py]: @agramfort 
Okay, so:
A
```python
np.any( np.array(freqs) <= 0):
```
B
```python
freqs = np.array(freqs)
if np.any(freqs <= 0):
```
Or C?
Make sure users only can pass ndarrays (which is already true for ```tfr_morlet``` and ```tfr_multitaper``` but not for ```morlet```)

**Comment by agramfort** [comment]
B

>

File: mne/time_frequency/tfr.py (hunk 1 above)
Comment (wmvanvliet) [reply mne/time_frequency/tfr.py]: I actually have a use case for only wanting to compute morlet (and multitaper I guess) at only one frequency. Better to cast to an array than fail in this case, I'd say.

File: mne/time_frequency/tfr.py [hunk 2]
```diff
@@ -66,6 +66,11 @@ def morlet(sfreq, freqs, n_cycles=7.0, sigma=None, zero_mean=False):
     Ws = list()
     n_cycles = np.atleast_1d(n_cycles)
 
+    freqs = np.array(freqs)
+    if np.any(freqs <= 0):
+        raise ValueError("all computed frequencies in freqs must be "
```
Comment (wmvanvliet) [inline mne/time_frequency/tfr.py]: What do you mean by "computed frequencies"? How about:

```
All frequencies in "freqs" must be greater than 0.
```

File: mne/time_frequency/tfr.py [hunk 3]
```diff
@@ -120,6 +125,12 @@ def _make_dpss(sfreq, freqs, n_cycles=7., time_bandwidth=4.0, zero_mean=False):
         The wavelets time series.
     """
     Ws = list()
+
+    freqs = np.array(freqs)
+    if np.any(freqs <= 0):
+        raise ValueError("all computed frequencies in freqs must be "
+                         "greater than 0.")
+
```
Comment (wmvanvliet) [inline mne/time_frequency/tfr.py]: Same here.

File: mne/time_frequency/tfr.py (hunk 2 above)
Comment (DiGyt) [reply mne/time_frequency/tfr.py]: Done!

File: mne/time_frequency/tfr.py (hunk 3 above)
Comment (DiGyt) [reply mne/time_frequency/tfr.py]: Done!

**Comment by cbrnr** [comment]
It's good that there is an error message, but I still don't know if these wavelet-based methods cannot compute the power at DC or if this is just the case in our implementation.

**Comment by DiGyt** [comment]
> It's good that there is an error message, but I still don't know if these wavelet-based methods cannot compute the power at DC or if this is just the case in our implementation.

Oh yes, sorry! I will work on this and do another PR if i find a convenient way. Would that be okay?

File: mne/time_frequency/tfr.py (hunk 1 above)
Comment (DiGyt) [reply mne/time_frequency/tfr.py]: Done!

**Comment by cbrnr** [comment]
Sure!

**Comment by agramfort** [comment]
thx @DiGyt

**Pull request merged**